app/Helpers/MoneyHelper.php

```diff
@@ -40,14 +43,128 @@ public static function format($amount, Currency $currency = null)
 
-        $money = new Money($amount * $minorUnitAdjustment, $moneyCurrency);
+        return $moneyFormatter->format($money);
+    }
 
-        $numberFormatter = new \NumberFormatter(App::getLocale(), \NumberFormatter::CURRENCY);
-        $moneyFormatter = new IntlMoneyFormatter($numberFormatter, $currencies);
+    /**
+     * Format a monetary amount (without the currency).
+     *
+     * Amount must be an integer, in exchange format.
+     * i.e. '100' for 1,00€
+     *
+     * @param int|null $amount
+     * @param Currency|int|null $currency
+     * @return string
+     */
+    public static function formatValue($amount, $currency = null): string
+    {
+        $currency = self::getCurrency($currency);
+
+        if (! $currency) {
+            return (string) ($amount / 100);
+        }
+
+        $moneyCurrency = new MoneyCurrency($currency->iso);
+        $money = new Money($amount ?? 0, $moneyCurrency);
+        $numberFormatter = new \NumberFormatter(App::getLocale(), \NumberFormatter::PATTERN_DECIMAL);
+        $moneyFormatter = new IntlMoneyFormatter($numberFormatter, new ISOCurrencies());
 
         return $moneyFormatter->format($money);
     }
+
+    /**
+     * Format a monetary amount in exchange value with currency symbol.
+     *
+     * @param float|null $exchange
+     * @param Currency|int|null $currency
+     * @return string
+     */
+    public static function display($exchange, $currency = null): string
+    {
+        $currency = self::getCurrency($currency);
+
+        return self::format(self::formatInput($exchange, $currency), $currency);
+    }
+
+    /**
+     * Format a monetary exchange value as storable integer.
+     *
+     * @param mixed|null $exchange
+     * @param Currency|int|null $currency
+     * @return int
+     */
+    public static function formatInput($exchange, $currency): int
+    {
+        $currency = self::getCurrency($currency);
+
+        if (! $currency) {
+            return (int) ((float) $exchange * 100);
+        }
+
+        $moneyParser = new DecimalMoneyParser(new ISOCurrencies());
+        $money = $moneyParser->parse((string) $exchange, $currency->iso);
+
+        return (int) $money->getAmount();
+    }
+
+    /**
+     * Format a monetary value as exchange value.
+     *
+     * @param int|null $amount
+     * @param Currency|int|null $currency
+     * @return float
+     */
+    public static function exchangeValue($amount, $currency): float
+    {
+        $currency = self::getCurrency($currency);
+
+        $moneyCurrency = new MoneyCurrency($currency->iso);
+        $money = new Money($amount ?? 0, $moneyCurrency);
+        $moneyFormatter = new DecimalMoneyFormatter(new ISOCurrencies());
+
+        return (float) $moneyFormatter->format($money);
+    }
+
+    /**
+     * Get unit adjustement value for the currency.
+     *
+     * @param Currency|int|null $currency
+     * @return int
+     */
+    public static function unitAdjustment($currency): int
```

Comment:
I also don't understand this.

Reply:
This method allows to convert from display value to exchange value giving the adjustment value needed to convert.
So for € or $, it will return 100 (for 2 digits precision), for Dinar, it will return 1000 (3 digits ...)
So you can divide your amount in exchange value by this number to convert (or otherwise ...).

And if you see, this was already used previously mixed in the `format` function.